tests/integration/_experimental/object_detection/test_evaluator_single_class_test_case_metrics.py

```diff
@@ -0,0 +1,910 @@
+        "iou=0.33 and same labels but 0.5 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.3",
+        "iou=0.33 and same labels but 0.5 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: F1-Optimal, IoU: 0.5, confidence ≥ 0.1",
+        "iou=0.33 and same labels but 0.5 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.3), IoU: 0.3, confidence ≥ 0.0",
+        "iou=0.33 and same labels but 0.99 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.0",
+        "iou=0.33 and same labels but 0.99 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.3",
+        "iou=0.33 and same labels but 0.99 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: F1-Optimal, IoU: 0.5, confidence ≥ 0.1",
+        "iou=0.33 and same labels but 0.99 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=1, TP=0, FN=1, FP=1, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.3), IoU: 0.3, confidence ≥ 0.0",
+        "iou=0.5 and same labels but 0 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=0, TP=0, FN=1, FP=0, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.0",
+        "iou=0.5 and same labels but 0 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=0, TP=0, FN=1, FP=0, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.5), IoU: 0.5, confidence ≥ 0.3",
+        "iou=0.5 and same labels but 0 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=0, TP=0, FN=1, FP=0, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: F1-Optimal, IoU: 0.5, confidence ≥ 0.1",
+        "iou=0.5 and same labels but 0 confidence",
+    ): TestCaseMetricsSingleClass(Objects=1, Inferences=0, TP=0, FN=1, FP=0, Precision=0.0, Recall=0.0, F1=0.0, AP=0.0),
+    (
+        "Threshold: Fixed(0.3), IoU: 0.3, confidence ≥ 0.0",
+        "iou=0.5 and same labels but 0.5 confidence",
+    ): TestCaseMetricsSingleClass(
+        Objects=1,
+        Inferences=1,
+        TP=1,
+        FN=0,
+        FP=0,
+        Precision=1.0,
+        Recall=1.0,
+        F1=1.0,
+        AP=0.0,
```

Comment:
AP should follow the PR curves if they exist. If a PR curve does not exist, maybe AP should be None instead of 0.0
Will think about this a bit more

Reply:
cc @gordonhart
if a PR curve does not exist, would it be better to see an AP of 0, the precision value, or None?
I think None is the best choice here

Reply:
Obviously this is a very rare edge case that I don't expect customers to come across but I personally think that the PR curve should be (0.0, 1.0) (i.e. extending the first precision value all the way to the y-axis)  and (1.0, 1.0) which should also give you `AP` value of 1.  But given that this is a very rare case and unlikely that anyone will stumble upon, I'm ok not changing the logic for it. 

Just to confirm though, let's say there are more than 1 point on the PR curve but recall doesn't exist all the way to 0 (so say we have (0.3, 0.5) and (0.7, 0.4)) then what would be the AP?  Would it be (0.5 * 0.3) + (0.4 * 0.4) = 0.31?

Reply:
It would be 0.29 by our sum of rectangular areas starting from (0,0), and precision=0 where recall is not defined (on the right)
```
(0.3 - 0) * 0.7 + (0.5 - 0.3) * 0.4 + (1 - 0.5) * 0
or
(0.4 - 0) * 0.5 + (0.7 - 0.4) * 0.3 + (1 - 0.7) * 0
```